File: test/utils/fixtures.go
package utils

import (
	"fmt"

	"github.com/lspecian/maas-mcp-server/internal/models"
)

// CreateTestMachine creates a test machine with the given parameters
func CreateTestMachine(systemID, hostname, status string) models.Machine {
	return models.Machine{
		SystemID:     systemID,
		Hostname:     hostname,
		FQDN:         hostname + ".example.com",
		Status:       status,
		StatusName:   status,
		Architecture: "amd64/generic",
		PowerState:   "off",
		PowerType:    "ipmi",
		Zone:         "default",
		Pool:         "default",
		Tags:         []string{"test"},
		IPAddresses:  []string{"192.168.1.100"},
		CPUCount:     4,
		Memory:       8192,
		OSSystem:     "ubuntu",
		DistroSeries: "focal",
		ResourceURL:  "/MAAS/api/2.0/machines/" + systemID + "/",
		Owner:        "",
		Description:  "Test machine",
	}
}

// CreateTestMachineWithDetails creates a test machine with network interfaces and block devices
func CreateTestMachineWithDetails(systemID, hostname, status string) models.Machine {
	machine := CreateTestMachine(systemID, hostname, status)

	// Add network interfaces
	machine.Interfaces = []models.NetworkInterface{
		{
			ID:          1,
			Name:        "eth0",
			Type:        "physical",
			Enabled:     true,
			MACAddress:  "00:11:22:33:44:55",
			VLANid:      1,
			Tags:        []string{"test-interface"},
			ResourceURL: "/MAAS/api/2.0/nodes/" + systemID + "/interfaces/1/",
			Links: []models.LinkInfo{
				{
					ID:        1,
					Mode:      "static",
					SubnetID:  1,
					IPAddress: "192.168.1.100",
				},
			},
		},
	}

	// Add block devices
	machine.BlockDevices = []models.BlockDevice{
		{
			ID:            1,
			Name:          "sda",
			Type:          "physical",
			Path:          "/dev/sda",
			Size:          107374182400, // 100 GB
			UsedSize:      107374182400,
			AvailableSize: 0,
			Model:         "Test Disk",
			Serial:        "TEST123",
			IDPath:        "/dev/disk/by-id/test-disk",
			Tags:          []string{"test-disk"},
			ResourceURL:   "/MAAS/api/2.0/nodes/" + systemID + "/blockdevices/1/",
		},
	}

	return machine
}

// CreateTestSubnet creates a test subnet with the given parameters
func CreateTestSubnet(id int, name, cidr string) models.Subnet {
	return models.Subnet{
		ID:          id,
		Name:        name,
		CIDR:        cidr,
		VLANid:      1,
		Space:       "default",
		GatewayIP:   "192.168.1.1",
		DNSServers:  []string{"8.8.8.8", "8.8.4.4"},
		Managed:     true,
		Active:      true,
		AllowDNS:    true,
		AllowProxy:  true,
		ResourceURL: fmt.Sprintf("/MAAS/api/2.0/subnets/%d/", id),
		FabricID:    1,
		FabricName:  "fabric-1",
		Description: "Test subnet",
	}
}

// CreateTestVLAN creates a test VLAN with the given parameters
func CreateTestVLAN(id, vid int, name string) models.VLAN {
	return models.VLAN{
		ID:          id,
		Name:        name,
		VID:         vid,
		MTU:         1500,
		FabricID:    1,
		FabricName:  "fabric-1",
		DHCPOn:      true,
		Primary:     false,
		ResourceURL: fmt.Sprintf("/MAAS/api/2.0/vlans/%d/", id),
		Description: "Test VLAN",
	}
}

// CreateTestTag creates a test tag with the given parameters
func CreateTestTag(name, description string) models.Tag {
	return models.Tag{
		Name:        name,
		Description: description,
		Definition:  "",
		Comment:     description,
		ResourceURL: "/MAAS/api/2.0/tags/" + name + "/",
	}
}

// CreateTestBlockDevice creates a test block device with the given parameters
func CreateTestBlockDevice(id int, name, path string, size int64) models.BlockDevice {
	return models.BlockDevice{
		ID:            id,
		Name:          name,
		Type:          "physical",
		Path:          path,
		Size:          size,
		UsedSize:      0,
		AvailableSize: size,
		Model:         "Test Disk",
		Serial:        fmt.Sprintf("TEST%d", id),
		IDPath:        fmt.Sprintf("/dev/disk/by-id/test-disk-%d", id),
		Tags:          []string{"test-disk"},
		ResourceURL:   fmt.Sprintf("/MAAS/api/2.0/blockdevices/%d/", id),
	}
}

// CreateTestNetworkInterface creates a test network interface with the given parameters
func CreateTestNetworkInterface(id int, name, macAddress string) models.NetworkInterface {
	return models.NetworkInterface{
		ID:          id,
		Name:        name,
		Type:        "physical",
		Enabled:     true,
		MACAddress:  macAddress,
		VLANid:      1,
		Tags:        []string{"test-interface"},
		Parents:     []int{},
		Children:    []int{},
		ResourceURL: fmt.Sprintf("/MAAS/api/2.0/interfaces/%d/", id),
	}
}

// CreateTestLinkInfo creates a test link info with the given parameters
func CreateTestLinkInfo(id int, mode, ipAddress string, subnetID int) models.LinkInfo {
	return models.LinkInfo{
		ID:        id,
		Mode:      mode,
		SubnetID:  subnetID,
		IPAddress: ipAddress,
	}
}

// CreateTestPartition creates a test partition with the given parameters
func CreateTestPartition(id int, path string, size int64) models.Partition {
	return models.Partition{
		ID:          id,
		Size:        size,
		UUID:        fmt.Sprintf("test-uuid-%d", id),
		Path:        path,
		Type:        "primary",
		ResourceURL: fmt.Sprintf("/MAAS/api/2.0/partitions/%d/", id),
	}
}

// CreateTestFilesystem creates a test filesystem with the given parameters
func CreateTestFilesystem(id int, fsType, mountPoint string) models.Filesystem {
	return models.Filesystem{
		ID:           id,
		UUID:         fmt.Sprintf("test-uuid-%d", id),
		FSType:       fsType,
		MountPoint:   mountPoint,
		MountOptions: "defaults",
		ResourceURL:  fmt.Sprintf("/MAAS/api/2.0/filesystems/%d/", id),
	}
}

File: src/test-utils/fixtures.ts
/**
 * Test fixtures for TypeScript tests
 */

/**
 * Create a machine fixture with the given parameters
 * @param id - Machine ID
 * @param hostname - Machine hostname
 * @param status - Machine status (default: 'Ready')
 * @returns Machine object
 */
export function createMachineFixture(id: string, hostname: string, status: string = 'Ready'): any {
  return {
    id,
    hostname,
    status,
    power_state: 'off',
    tags: ['test'],
    interfaces: [
      { 
        id: 'eth0', 
        name: 'eth0', 
        mac_address: '00:11:22:33:44:55', 
        ip_address: '192.168.1.100' 
      },
    ],
    storage: [
      { 
        id: 'sda', 
        name: 'sda', 
        size: 1000000000, 
        type: 'ssd' 
      },
    ],
  };
}

/**
 * Create a machine details fixture with the given parameters
 * @param id - Machine ID
 * @param hostname - Machine hostname
 * @param status - Machine status (default: 'Ready')
 * @returns Machine details object
 */
export function createMachineDetailsFixture(id: string, hostname: string, status: string = 'Ready'): any {
  return {
    id,
    hostname,
    fqdn: `${hostname}.example.com`,
    status,
    architecture: 'amd64/generic',
    power_state: 'off',
    power_type: 'ipmi',
    zone: 'default',
    pool: 'default',
    tags: ['test'],
    ip_addresses: ['192.168.1.100'],
    cpu_count: 4,
    memory: 8192,
    os_system: 'ubuntu',
    distro_series: 'focal',
    interfaces: [
      { 
        id: 'eth0', 
        name: 'eth0', 
        type: 'physical',
        enabled: true,
        mac_address: '00:11:22:33:44:55', 
        ip_address: '192.168.1.100',
        vlan_id: 1,
        links: [
          {
            mode: 'static',
            subnet_id: 1,
            ip_address: '192.168.1.100'
          }
        ]
      },
    ],
    storage: [
      { 
        id: 'sda', 
        name: 'sda', 
        type: 'physical',
        path: '/dev/sda',
        size: 107374182400, // 100 GB
        used_size: 107374182400,
        available_size: 0,
        model: 'Test Disk',
        serial: 'TEST123'
      },
    ],
    owner: '',
    description: 'Test machine'
  };
}

/**
 * Create a subnet fixture with the given parameters
 * @param id - Subnet ID
 * @param name - Subnet name
 * @param cidr - Subnet CIDR
 * @returns Subnet object
 */
export function createSubnetFixture(id: string, name: string, cidr: string): any {
  return {
    id,
    name,
    cidr,
    fabric: 'fabric-1',
    vlan: 'vlan-1',
    vlan_tag: 1,
    space: 'default',
    gateway_ip: '192.168.1.1',
    dns_servers: ['8.8.8.8', '8.8.4.4'],
    managed: true,
    active: true,
    ip_ranges: [
      {
        type: 'dynamic',
        start_ip: '192.168.1.100',
        end_ip: '192.168.1.200'
      }
    ]
  };
}

/**
 * Create a VLAN fixture with the given parameters
 * @param id - VLAN ID
 * @param name - VLAN name
 * @param vid - VLAN VID
 * @returns VLAN object
 */
export function createVLANFixture(id: string, name: string, vid: number): any {
  return {
    id,
    name,
    vid,
    mtu: 1500,
    fabric: 'fabric-1',
    fabric_id: '1',
    dhcp_enabled: true,
    primary: false
  };
}

/**
 * Create a tag fixture with the given parameters
 * @param name - Tag name
 * @param description - Tag description
 * @returns Tag object
 */
export function createTagFixture(name: string, description: string): any {
  return {
    name,
    description,
    definition: '',
    comment: description
  };
}

/**
 * Create a block device fixture with the given parameters
 * @param id - Block device ID
 * @param name - Block device name
 * @param path - Block device path
 * @param size - Block device size in bytes
 * @returns Block device object
 */
export function createBlockDeviceFixture(id: string, name: string, path: string, size: number): any {
  return {
    id,
    name,
    type: 'physical',
    path,
    size,
    used_size: 0,
    available_size: size,
    model: 'Test Disk',
    serial: `TEST${id}`,
    tags: ['test-disk']
  };
}

/**
 * Create a network interface fixture with the given parameters
 * @param id - Network interface ID
 * @param name - Network interface name
 * @param macAddress - Network interface MAC address
 * @returns Network interface object
 */
export function createNetworkInterfaceFixture(id: string, name: string, macAddress: string): any {
  return {
    id,
    name,
    type: 'physical',
    enabled: true,
    mac_address: macAddress,
    vlan_id: 1,
    tags: ['test-interface'],
    links: [
      {
        mode: 'static',
        subnet_id: 1,
        ip_address: '192.168.1.100'
      }
    ]
  };
}

/**
 * Create a pagination fixture with the given parameters
 * @param items - Array of items
 * @param total - Total number of items
 * @param page - Current page number
 * @param perPage - Number of items per page
 * @returns Pagination object
 */
export function createPaginationFixture(items: any[], total: number, page: number, perPage: number): any {
  return {
    items,
    total,
    page,
    per_page: perPage
  };
}

/**
 * Create an error fixture with the given parameters
 * @param code - Error code
 * @param message - Error message
 * @returns Error object
 */
export function createErrorFixture(code: string, message: string): any {
  return {
    error: {
      code,
      message
    }
  };
}

/**
 * Create a success response fixture with the given data
 * @param data - Response data
 * @returns Success response object
 */
export function createSuccessFixture(data: any): any {
  return {
    success: true,
    data
  };
}

/**
 * Create a mock MCP tool response
 * @param toolName - Name of the tool
 * @param result - Tool result
 * @returns MCP tool response object
 */
export function createMCPToolResponseFixture(toolName: string, result: any): any {
  return {
    tool: toolName,
    result
  };
}

/**
 * Create a mock MCP resource response
 * @param uri - Resource URI
 * @param data - Resource data
 * @returns MCP resource response object
 */
export function createMCPResourceResponseFixture(uri: string, data: any): any {
  return {
    uri,
    data
  };
}